org.lflang/src/org/lflang/generator/python/PythonGenerator.java

```diff
@@ -794,19 +795,25 @@ public void generateStateVariableInitializations(ReactorInstance instance) {
      * @param instance The reactor instance.
      */
     @Override
-    public void generateParameterInitialization(ReactorInstance instance) {
+    protected void generateParameterInitialization(ReactorInstance instance) {
         // Do nothing
         // Parameters are initialized in Python
     }
 
+    @Override
+    protected void generateMethods(ReactorDecl reactor) {
+        // Do nothing.
+        // Methods are generated in Python not C.
```

Comment:
This is overriding a method in the CGenerator that is responsible for generating methods in C

Reply:
But why isn't that happening in the body of this method then?

Reply:
The CGenerator is responsible for putting things in a Foo.c file (for Foo.lf) and the PythonGenerator is responsible for putting things in a Foo.py file. There is some explanation [here](https://www.lf-lang.org/docs/handbook/target-language-details#target-implementation-details) about what each generated file is.

The overriden method here is called while the CGenerator is filling up Foo.c.

Reply:
> Is there a technical reason for this?

There is a clear separation in the Python target for where C-based code generation occurs and where Python-based code generation occurs. In `PythonGenerator.doGenerate()`, the call to `super.doGenerate()` will result in the creation of pure C code. After that, the rest of the logic is dedicated to generating `.py` code. However, some aspects of the inherited CGenerator (that produces the C code) are modified slightly (by overriding its methods) in the PythonGenerator. For example, methods must be ignored by the CGenerator. Still, the overriden methods still produce pure C code if applicable, or do nothing if no pure C code needs to be generated.

> I think it would be helpful to revise the comment and point to the location where the method is eventually generated.

Okay. Done [here](https://github.com/lf-lang/lingua-franca/pull/1230/commits/210c15476da16c6c98503d37901211dd7795369e).